### ISSUE TITLE
super in object literals

### ISSUE THREAD

**Issue by suyashv** [6to5/6to5#411]
```js
let a = {
  name () {
    return 'Suyash';
  }
};

let b = {
  name () {
    return super() + ' Verma';
  }
};

Object.setPrototypeOf(b, a);

console.log(b.name());
```

compiles to 

```js
"use strict";

var a = {
  name: function name() {
    return "Suyash";
  }
};

var b = {
  name: function name() {
    return super() + " Verma";
  }
};

Object.setPrototypeOf(b, a);

console.log(b.name());
```

which generates an error

```shell
    return super() + " Verma";
           ^^^^^
SyntaxError: Unexpected reserved word
```

**Comment by sebmck** [comment]
`super()` currently isn't supported in object literals. It'll be trivial to add after #411 is merged however.

**Comment by thejameskyle** [comment]
@sebmck This _is_ #411.

**Comment by sebmck** [comment]
Woops... Meant #408

**Comment by fkling** [comment]
I think it's worth pointing out that`super()` only works inside a `constructor` method. In all other methods it has to be `super.methodName()`.

**Comment by stefanpenner** [comment]
> return super() + " Verma";

also, isn't supported syntax. currently it would be `super.methodName()`

If someone is interested, they should skim the `homeObject` stuff in the spec. This allows for `super` that doesn't always walk the logical prototype chain, thereby enabling super through a series of mixins that are merged into a single prototype level.

Polyfill of `toMethod` and the homeObject concept is complicated, but likely possible.

I have an example of various `super` techniques https://github.com/stefanpenner/perf-scratch/blob/master/super.js#L79-L87
